src/plone/exportimport/utils/principals/members.py

```diff
@@ -78,8 +78,8 @@ def _get_base_user_data(member: MemberData):
     # username, groups, roles
     props = {
         "username": user_id,
-        "roles": json_compatible(roles),
-        "groups": json_compatible(groups),
+        "roles": json_compatible(sorted(roles)),
```

Comment:
It's already sorted by helpers.get_roles_for_member

Reply:
Agreed, let's sort only once.  Done.